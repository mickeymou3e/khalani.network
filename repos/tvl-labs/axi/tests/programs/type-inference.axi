function f(x: Nat) -> Boolean;

function g(x: Bytes) -> Boolean;

proc p(x) { f(x) }

forall x. p(x);

forall x, y. (x = y & p(y));

exists x, y. (x = y & p(x));

forall x. ((exists y. y = x) & p(x));

struct Point {
    x: Nat,
    y: Nat,
}

function fp(p: Point) -> Boolean;

forall p. (fp(p) & p.x = 1);

# Type error for g(y).
forall x, y. (x = y & p(x) & g(y));
