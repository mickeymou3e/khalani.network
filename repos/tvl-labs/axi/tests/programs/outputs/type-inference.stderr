[31mError:[0m 
    [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m<unknown>:25:32[38;5;246m][0m
    [38;5;246mâ”‚[0m
 [38;5;246m25 â”‚[0m [38;5;249mf[0m[38;5;249mo[0m[38;5;249mr[0m[38;5;249ma[0m[38;5;249ml[0m[38;5;249ml[0m[38;5;249m [0m[38;5;249mx[0m[38;5;249m,[0m[38;5;249m [0m[38;5;249my[0m[38;5;249m.[0m[38;5;249m [0m[38;5;249m([0m[38;5;249mx[0m[38;5;249m [0m[38;5;249m=[0m[38;5;249m [0m[38;5;249my[0m[38;5;249m [0m[38;5;249m&[0m[38;5;249m [0m[38;5;249mp[0m[38;5;249m([0m[38;5;249mx[0m[38;5;249m)[0m[38;5;249m [0m[38;5;249m&[0m[38;5;249m [0m[38;5;249mg[0m[38;5;249m([0my[38;5;249m)[0m[38;5;249m)[0m[38;5;249m;[0m
 [38;5;240m   â”‚[0m                                â”¬  
 [38;5;240m   â”‚[0m                                â•°â”€â”€ failed to unify
[38;5;246mâ”€â”€â”€â”€â•¯[0m
